### PR TITLE
stats2

### PR DESCRIPTION
**Pull request by emomaxd**

### CODE REVIEW

File: editor-components/stats/stats2/stats2.tsx [hunk 1]
```diff
@@ -177,38 +130,113 @@ class Stats2Page extends BaseStats {
...
-                  <div className={this.decorateCSS("bottoms")}>
-                    <ComposerLink path={cardData.buttonLink}>
-                      <p className={this.decorateCSS("card-button-text")}>{cardData.buttonText}</p>
-                    </ComposerLink>
-                  </div>
-                </div>
-              )
-            )}
+          <div className={this.decorateCSS("header-wrapper")}>
+            <div className={this.decorateCSS("header")}>{this.getPropValue("header")}</div>
```
Comment (dogusmiuw) [inline editor-components/stats/stats2/stats2.tsx:139]: remove relevant if `this.getPropValue("header")` value is empty

[check this out](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/stats/stats2/stats2.tsx [hunk 2]
```diff
@@ -177,38 +130,113 @@ class Stats2Page extends BaseStats {
...
+            <div className={this.decorateCSS("subHeader")}>
+              {this.getPropValue("subHeader")}
```
Comment (dogusmiuw) [inline editor-components/stats/stats2/stats2.tsx:141]: remove relevant if `this.getPropValue("subHeader")` value is empty

[check this out](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/stats/stats2/stats2.tsx [hunk 3]
```diff
@@ -177,38 +130,113 @@ class Stats2Page extends BaseStats {
...
+              <button className={this.decorateCSS("contact-button")}>{this.getPropValue("contactButton")}</button>
```
Comment (dogusmiuw) [inline editor-components/stats/stats2/stats2.tsx:142]: remove button if `this.getPropValue("contactButton")` value is empty

[check this out](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/stats/stats2/stats2.tsx [hunk 4]
```diff
@@ -177,38 +130,113 @@ class Stats2Page extends BaseStats {
...
+            </div>
+          </div>
+
+          <div className={this.decorateCSS("cards-container")}>
+            {cards.map((card, index) => (
```
Comment (dogusmiuw) [inline editor-components/stats/stats2/stats2.tsx:147]:
```suggestion
            {cards.map((card: Card, index: number) => (
```

File: editor-components/stats/stats2/stats2.tsx [hunk 5]
```diff
@@ -177,38 +130,113 @@ class Stats2Page extends BaseStats {
...
+        <p className={this.props.styles["card-text"]}>{this.props.card.text}</p>
+        <p className={this.props.styles["card-amount"]}>
+          {digits.map((digit, index) => (
+            <span key={index} className={digit > 0 ? this.props.styles["digit"] : ""}>{digit}</span>
+          ))}
+        </p>
+      </div>
+    );
+  }
+}
```
Comment (dogusmiuw) [inline editor-components/stats/stats2/stats2.tsx:240]: you do not need to create another component for this. there is an example for number animation in stats6.

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
please handle changes

**Comment by dogusmiuw** [comment]
also this component doesnt look like the original one. please check carefully

File: editor-components/stats/stats2/stats2.tsx [hunk 6]
```diff
@@ -156,87 +166,60 @@ class Stats2Page extends BaseStats {
...
+    }, 30);
+  };
+
+  return (
+    <div ref={ref} className={styles["listed"]}>
+      <p className={styles["card-text"]}>{card.text}</p>
+      <p className={styles["card-amount"]}>{amount}</p>
+    </div>
+  );
+};
```
Comment (dogusmiuw) [inline editor-components/stats/stats2/stats2.tsx:223]: remove this we do not use functional components

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
do not use functional component

File: editor-components/stats/stats2/stats2.module.scss [hunk 7]
```diff
@@ -1,114 +1,245 @@
...
+        .listed {
+          display: flex;
+          width: 50%;
+          flex-direction: column;
+          justify-content: space-between;
+          flex-grow: 1;
+          padding: 40px 60px 30px 60px;
+          border-left: 1px solid rgba(var(--composer-font-color-primary-rgb), 0.2);
+          border-right: 1px solid rgba(var(--composer-font-color-primary-rgb), 0.2);
+          border-top: 1px solid rgba(var(--composer-font-color-primary-rgb), 0.2);
```
Comment (Ayazelar) [inline editor-components/stats/stats2/stats2.module.scss:86]:
```suggestion
      border: 1px solid rgba(var(--composer-font-color-primary-rgb), 0.2);
      border-bottom: none;
```

File: editor-components/stats/stats2/stats2.tsx [hunk 8]
```diff
@@ -177,32 +197,63 @@ class Stats2Page extends BaseStats {
...
+                    )}
+                  </button>
+                )}
+              </div>
+            )}
+
+            {cards.length > 0 && (
+              <div
+                className={this.decorateCSS("cards-container")}
+                style={{ top: hasHeader ? "245px" : "0" }}
```
Comment (Ayazelar) [inline editor-components/stats/stats2/stats2.tsx:246]: don't use inline style

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

File: editor-components/stats/stats2/stats2.tsx [hunk 9]
```diff
@@ -175,34 +195,78 @@ class Stats2Page extends BaseStats {
...
-                      <p className={this.decorateCSS("card-button-text")}>{cardData.buttonText}</p>
-                    </ComposerLink>
-                  </div>
-                </div>
-              )
+          {header && (
+            <div className={`${this.decorateCSS("header-wrapper")} ${
+              cardLength <= 0 ? this.decorateCSS("full-width") : ""
+            }`}>
+              <div className={this.decorateCSS("header")}>{header}</div>
```
Comment (Ayazelar) [inline editor-components/stats/stats2/stats2.tsx:216]: use this.getPropValue

File: editor-components/stats/stats2/stats2.tsx [hunk 10]
```diff
@@ -175,34 +195,78 @@ class Stats2Page extends BaseStats {
...
+          <div className={this.decorateCSS("bottom-content")}>
+            {(subHeader || contactButton || contactButtonIcon) && (
+               <div className={`${this.decorateCSS("subHeader")} ${
+                cardLength <= 0 ? this.decorateCSS("full-width") : ""
+              }`}>
+               {subHeader && (
+                 <div className={`${this.decorateCSS("description")} ${
+                  cardLength <= 0 ? this.decorateCSS("full-width") : ""
+                }`}>
+                 {subHeader}
```
Comment (Ayazelar) [inline editor-components/stats/stats2/stats2.tsx:229]: use getPropValue

File: editor-components/stats/stats2/stats2.tsx [hunk 11]
```diff
@@ -175,34 +195,78 @@ class Stats2Page extends BaseStats {
...
+                 </div>
+               )}
+               {(contactButton || contactButtonIcon) && (
+                 <button
+                   className={`${this.decorateCSS("contact-button")} ${
+                     buttonAnimationEnabled ? this.decorateCSS("animated") : ""
+                   } ${cardLength <= 0 ? this.decorateCSS("button-full-width") : ""
+                  }`}
+                 >
+                   {contactButton}
```
Comment (Ayazelar) [inline editor-components/stats/stats2/stats2.tsx:239]: use getPropValue

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

File: editor-components/stats/stats2/stats2.tsx [hunk 12]
```diff
@@ -211,4 +281,116 @@ class Stats2Page extends BaseStats {
...
+            )}
+            {(amount !== null && amount !== "NaN") && (
+              <h2 className={styles["card-amount"]}>
+                {integerPart}
+                {showDecimals && decimalPart && <span>.{decimalPart}</span>}
+              </h2>
+            )}
+            {card.secondIcon && (
+              <ComposerIcon
+                propsIcon={{ className: styles["card-icon-after"] }}
```
Comment (Ayazelar) [inline editor-components/stats/stats2/stats2.tsx:385]: use decorateCss

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

File: editor-components/stats/stats2/stats2.module.scss [hunk 13]
```diff
@@ -1,114 +1,265 @@
...
-                    border-radius: var(--composer-border-radius);
-                    padding: 5px 0 5px 30px;
-                    .card-button-text {
-                        color: var(--composer-secondary-color);
-                        font-weight: 600;
-                        cursor: pointer;
-                    }
-                }
+            justify-content: center;
+            padding: 1rem 2rem;
```
Comment (Ayazelar) [inline editor-components/stats/stats2/stats2.module.scss:53]: use gap varaible

File: editor-components/stats/stats2/stats2.module.scss [hunk 14]
```diff
@@ -1,114 +1,265 @@
...
+      }
+    }
+
+    .bottom-content {
+      flex-wrap: unset;
+
+      .subHeader {
+        flex: 1;
+        position: sticky;
+        top: 5rem;
```
Comment (Ayazelar) [inline editor-components/stats/stats2/stats2.module.scss:29]: when the font size changed the top position also will changed. is that expected?

File: editor-components/stats/stats2/stats2.module.scss [hunk 15]
```diff
@@ -1,114 +1,265 @@
...
+              transition: transform 0.4s ease;
+            }
+
+            &.animated:hover {
+              transform: translateY(-2px);
+            }
+
+            .contact-button-icon {
+              flex-shrink: 0;
+              margin-left: .5rem;
```
Comment (Ayazelar) [inline editor-components/stats/stats2/stats2.module.scss:73]: use gap value

File: editor-components/stats/stats2/stats2.module.scss [hunk 16]
```diff
@@ -1,114 +1,265 @@
...
+                max-width: 60%;
+              }
+            }
+          }
         }
+
+        .cards-container {
+          .listed {
+            width: 100%;
+            padding: 2rem 2.5rem;
```
Comment (Ayazelar) [inline editor-components/stats/stats2/stats2.module.scss:204]: use gap value

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

File: editor-components/stats/stats2/stats2.module.scss [hunk 17]
```diff
@@ -1,114 +1,241 @@
...
+
+      .cards-container {
+        gap: 0;
+        flex: 3;
+
+        .listed {
+          display: flex;
+          flex-direction: column;
+          border: 1px solid rgba(var(--composer-font-color-primary-rgb), 0.2);
+          padding: 1rem 3rem 1.5rem 3rem;
```
Comment (KardelennArslan) [inline editor-components/stats/stats2/stats2.module.scss:73]: use gap value

**Pull request merged**